docs/security_bots.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Managing Bots
+tags:
+  - Security
+  - Bots
+---
+
+# Managing Bots
+
+A public-facing production instance will likely encounter bad bots and other malicious traffic that will consume resources. There are many solutions available that address a variety of different needs, but we provide basic configurations and a Docker image for integrating the [NGINX Ultimate Bad Bot & Referrer Blocker](https://github.com/mitchellkrogza/nginx-ultimate-bad-bot-blocker).
+
+## Deployment
+
+1. Uncomment or add the following docker-compose environment variables, replacing any appropriate values with your own (leave the blocker and cron disabled to start):
+   ```dotenv title=".env"
+   MSMTP_ACCOUNT=SMTP_ACCOUNT_NAME
+   MSMTP_EMAIL=repositorysupport@metro.org
+   MSMTP_HOST=smtp.metro.org
+   MSMTP_PASSWORD=YOUR_SMTP_PASSWORD
+   MSMTP_PORT=SMTP_PORT
+   MSMTP_STARTTLS=on
+   NGXBLOCKER_ENABLE=false
+   NGXBLOCKER_CRON=00 22 * * *
+   NGXBLOCKER_CRON_COMMAND=/usr/local/sbin/update-ngxblocker -x
+   NGXBLOCKER_CRON_START=false
+   ```
+2. Uncomment or add the following lines and comment out the line for the original NGINX image:
+   ```yaml title="docker-compose.yml" hl_lines="7 8 15-24 33"
+   # Run docker-compose up -d
+   # Docker file for Arm64 and Apple M1 machines
+   version: '3.5'
+   services:
+     web:
+       container_name: esmero-web
+       # image: jonasal/nginx-certbot
+       image: esmero/nginx-bot-blocker:1.1.0-multiarch
+       restart: always
+       environment:
+         CERTBOT_EMAIL: ${ARCHIPELAGO_EMAIL}
+         ENVSUBST_VARS: FQDN
+         FQDN: ${ARCHIPELAGO_DOMAIN}
+         NGINX_ENVSUBST_OUTPUT_DIR: /etc/nginx/user_conf.d
+         MSMTP_ACCOUNT: ${MSMTP_ACCOUNT}
+         MSMTP_EMAIL: ${MSMTP_EMAIL}
+         MSMTP_HOST: ${MSMTP_HOST}
+         MSMTP_PASSWORD: ${MSMTP_PASSWORD}
+         MSMTP_PORT: ${MSMTP_PORT}
+         MSMTP_STARTTLS: ${MSMTP_STARTTLS}
+         NGXBLOCKER_CRON: ${NGXBLOCKER_CRON}
+         NGXBLOCKER_CRON_COMMAND: ${NGXBLOCKER_CRON_COMMAND}
+         NGXBLOCKER_CRON_START: ${NGXBLOCKER_CRON_START}
+         NGXBLOCKER_ENABLE: ${NGXBLOCKER_ENABLE}
+       ports:
+         - "80:80"
+         - "443:443"
+       volumes:
+         - ${ARCHIPELAGO_ROOT}/config_storage/nginxconfig/template:/etc/nginx/templates
+         - ${ARCHIPELAGO_ROOT}/drupal:/var/www/html:cached
+         - ${ARCHIPELAGO_ROOT}/data_storage/ngnixcache:/var/cache/nginx
+         - ${ARCHIPELAGO_ROOT}/data_storage/letsencrypt:/etc/letsencrypt
+         - ${ARCHIPELAGO_ROOT}/config_storage/nginxconfig/bots.d:/etc/nginx/bots.d
+   ```
+3. Bring the docker ensemble down, pull the new image, and bring the ensemble back up:
```

Comment:
To avoid downtimes, you can get/pull the image without bringing the docker-compose down. Once pulled, you can bring it down and then up

Reply:
Resolved via d8e1ad5.